File: src/main/java/com/kafkaworkshop/workshopdemo/UserService.java
package com.kafkaworkshop.workshopdemo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

public interface UserService {
    UserDto getUserIdByName(String userId);
    void registerUser(UserDto userDto);
}

File: src/main/resources/static/js/UserService.js
'use strict'
angular.module('demo.services', []).factory('UserService', ["$http", "CONSTANTS", function($http, CONSTANTS) {
    var service = {};
    service.getUserIdByName = function(userName) {
        var url = CONSTANTS.getUseIdUrl;
        return $http.get(url);
    }
    service.registerUser = function(userDto) {
        return $http.post(CONSTANTS.registerUser, userDto);
    }
    return service;
}]);